flink-java/src/main/java/org/apache/flink/api/java/operators/DataSink.java

```diff
@@ -208,6 +214,28 @@ public DataSink(DataSet<T> data, OutputFormat<T> format, TypeInformation<T> type
 		return this;
 	}
 
+	private void isValidSortKeyType(int field) {
```

Comment:
I agree, there is some duplicated code for key handling, not only for sort keys. 
I am planning to rework the expression key handling in the API to make it more consistent as of FLINK-1666 soon.